### PR TITLE
Add capnp serialization to RecordSensor region

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2835.

### CODE REVIEW

**Comment by chetan51** [comment]
@scottpurdy

**Comment by scottpurdy** [comment]
Is this ready for review?

**Comment by chetan51** [comment]
@scottpurdy Yes.

File: src/nupic/regions/RecordSensor.py [hunk 1]
```diff
@@ -511,3 +519,37 @@ def setParameter(self, parameterName, index, parameterValue):
...
+    """
+    regionImpl = proto.regionImpl.as_struct(RecordSensorProto)
+
+    self.encoder.write(regionImpl.encoder)
+    self.disabledEncoder.write(regionImpl.disabledEncoder)
+    regionImpl.topDownMode = self.topDownMode
+    regionImpl.verbosity = self.verbosity
+    regionImpl.numCategories = self.numCategories
+
+    # TODO: Write preEncodingFilters and postEncodingFilters
```
Comment (scottpurdy) [inline src/nupic/regions/RecordSensor.py]: ?

Comment (chetan51) [inline src/nupic/regions/RecordSensor.py]: `preEncodingFilters` and `postEncodingFilters` are not actually being used anywhere. Should we just remove them and not serialize them?

Comment (scottpurdy) [inline src/nupic/regions/RecordSensor.py]: Yeah probably for now. I'm doubt that functionlaity would worked if we tried to use it.

**Comment by scottpurdy** [comment]
:+1: after removing the TODO

**Pull request merged**